src/lib.rs

```diff
@@ -596,27 +596,40 @@ where
 {
     let samples: Vec<_> = items
         .into_iter()
+        .take(t)
         .map(|(i, sample)| (into_fr_plus_1(i), sample))
         .collect();
     if samples.len() < t {
         return Err(Error::NotEnoughShares);
     }
+
+    // Compute the products `x_prod[i]` of all but the `i`-th entry.
+    let mut x_prod: Vec<C::Scalar> = Vec::with_capacity(t);
+    let mut tmp = C::Scalar::one();
+    x_prod.push(tmp);
+    for (x, _) in samples.iter().take(t - 1) {
+        tmp.mul_assign(x);
+        x_prod.push(tmp);
+    }
+    let mut tmp = C::Scalar::one();
```

Comment:
This line could be `tmp = C::Scalar::one();` in the spirit of imperative algorithms.

Reply:
I amended the commit:

```diff
-    let mut tmp = C::Scalar::one();
+    tmp = C::Scalar::one();
```